include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2477,47 +3160,1289 @@ namespace internal
     }
   };
 
-
-
-  template <int dim, int fe_degree, typename Number, bool lex_faces = false>
-  struct FEFaceNormalEvaluationImpl
+  template <bool symmetric_evaluate,
+            int  dim,
+            int  fe_degree,
+            int  n_q_points_1d,
+            typename Number>
+  struct FEFaceEvaluationImplRaviartThomas
   {
-    template <bool do_evaluate, bool add_into_output>
-    static void
-    interpolate(const unsigned int                            n_components,
-                const EvaluationFlags::EvaluationFlags        flags,
-                const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
-                const Number *                                input,
-                Number *                                      output,
-                const unsigned int                            face_no)
+    // We enable a transformation to collocation for derivatives if it gives
+    // correct results (first two conditions), if it is the most efficient
+    // choice in terms of operation counts (third condition) and if we were
+    // able to initialize the fields in shape_info.templates.h from the
+    // polynomials (fourth condition).
+    static_assert(
+      !symmetric_evaluate,
+      "symmetric_evaluate = true is not yet implemented for Raviart-Thomas");
+
+    using Eval_normal  = EvaluatorTensorProduct<evaluate_raviart_thomas,
+                                               dim - 1,
+                                               std::max(fe_degree + 1, 1),
+                                               n_q_points_1d,
+                                               Number>;
+    using Eval_tangent = EvaluatorTensorProduct<evaluate_raviart_thomas,
+                                                dim - 1,
+                                                std::max(fe_degree, 1),
+                                                n_q_points_1d,
+                                                Number>;
+
+    using Eval_general = EvaluatorTensorProduct<evaluate_general,
+                                                dim - 1,
+                                                std::max(fe_degree, 0),
+                                                n_q_points_1d,
+                                                Number>;
+    template <typename Eval_type>
+    static Eval_type
+    create_evaluator_tensor_product(
+      const MatrixFreeFunctions::UnivariateShapeData<Number> &data,
+      const unsigned int                                      subface_index,
+      const unsigned int                                      direction)
     {
-      Assert(static_cast<unsigned int>(fe_degree) ==
-                 shape_info.data.front().fe_degree ||
-               fe_degree == -1,
-             ExcInternalError());
+      if (symmetric_evaluate) // TODO
```

Comment:
Could you go through the TODO and add a comment what the problem there is (and maybe also note these an issue).

Reply:
I've already remove all todos as Martin told me to, since one would have to go through everything again anyway to check what works and what doesn't.